### PR TITLE
Run build_android.sh in Jenkins

### PR DESCRIPTION
**Pull request by pietern**

### CODE REVIEW

File: scripts/build_android.sh [hunk 1]
```diff
@@ -19,58 +19,77 @@
...
+$CAFFE2_ROOT/scripts/build_host_protoc.sh
 
-# Now, actually build the android target.
-echo "Building caffe2"
+# Now, actually build the Android target.
+BUILD_ROOT=$CAFFE2_ROOT/build_android
+mkdir -p $BUILD_ROOT
 cd $BUILD_ROOT
 
+CMAKE_ARGS=()
```
Comment (Yangqing) [inline scripts/build_android.sh]: Interesting, this line is causing issues here:

https://travis-ci.org/caffe2/caffe2/jobs/302897319#L2002

But it should be correct...

**Comment by pietern** [comment]
Yeah... perhaps they run some super old version of bash...? I'll investigate.

**Comment by pietern** [comment]
Output on Jenkins (for android build):

```
00:01:10.495 Bash: GNU bash, version 4.3.48(1)-release (x86_64-pc-linux-gnu)
00:01:10.495 Caffe2 path: /var/lib/jenkins/workspace
00:01:10.495 Using Android NDK at /opt/ndk
```

**Comment by pietern** [comment]
Output on Travis:

```
Bash: GNU bash, version 4.3.11(1)-release (x86_64-pc-linux-gnu)
Caffe2 path: /home/travis/build/caffe2/caffe2
Using Android NDK at /opt/android_ndk
```

**Comment by pietern** [comment]
Missed a critical detail: Travis runs `sh ./.travis/build.sh` so it likely uses a minimal non-Bash shell.

**Comment by pietern** [comment]
We don't run tests for the Android build, which is why it was failing. I updated the configuration to allow a test build to finish without test result files.

@caffe2bot retest this please

**Comment by pietern** [comment]
@Yangqing This is now good to go.

**Comment by pietern** [comment]
Rebased to fix merge conflict

**Comment by pietern** [comment]
Unrelated Travis failure (see #1504).

**Pull request closed without merging**